ios/PaymentSheetAppearance.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  PaymentSheetAppearance.swift
+//  stripe-react-native
+//
+//  Created by Charles Cruzan on 5/11/22.
+//
+@_spi(STP) import Stripe
+
+extension StripeSdk {
+    func buildPaymentSheetAppearance(userParams: NSDictionary) throws -> PaymentSheet.Appearance {
+        var appearance = PaymentSheet.Appearance()
+        
+        if let fontParams = userParams["font"] as? NSDictionary {
+            appearance.font = try buildFont(params: fontParams)
+        }
+        if let colorParams = userParams["colors"] as? NSDictionary {
+            appearance.colors = try buildColors(params: colorParams)
+        }
+        if let shapeParams = userParams["shapes"] as? NSDictionary {
+            appearance.cornerRadius = shapeParams["borderRadius"] as? CGFloat ?? PaymentSheet.Appearance.default.cornerRadius
+            appearance.borderWidth = shapeParams["borderWidth"] as? CGFloat ?? PaymentSheet.Appearance.default.borderWidth
+            if let shadowParams = shapeParams["shadow"] as? NSDictionary {
+                appearance.shadow = buildShadow(params: shadowParams)
+            }
+        }
+        if let primaryButtonParams = userParams["primaryButton"] as? NSDictionary {
+            appearance.primaryButton = try buildPrimaryButton(params: primaryButtonParams)
+        }
+        
+        return appearance
+    }
+    
+    private func buildFont(params: NSDictionary) throws -> Stripe.PaymentSheet.Appearance.Font {
+        var font = Stripe.PaymentSheet.Appearance.Font()
+        if let fontName = params["family"] as? String {
+            guard let customFont = UIFont(name: fontName, size: UIFont.systemFontSize) else {
+                throw PaymentSheetAppearanceError.missingFont(fontName)
+            }
+            font.base = customFont
+        }
+        font.sizeScaleFactor = params["scale"] as? CGFloat ?? PaymentSheet.Appearance.default.font.sizeScaleFactor
+        return font
+    }
+    
+    private func buildColors(params: NSDictionary) throws -> Stripe.PaymentSheet.Appearance.Colors {
+        var colors = Stripe.PaymentSheet.Appearance.Colors()
+        
+        if (params.object(forKey: "light") != nil && params.object(forKey: "dark") == nil ||
+            params.object(forKey: "dark") != nil && params.object(forKey: "light") == nil) {
+            throw PaymentSheetAppearanceError.missingAppearanceMode
+        }
```

Comment:
nit: Can this just be:
```swift
if params.object(forKey: "light") == nil || params.object(forKey: "dark") == nil {
```

Reply:
unfortunately no 😞 it's possible for both to be null if the merchant provides the base `colors` config, in lieu of distinct `colors.light` and `colors.dark` configs.